creates ui and functionality for background jobs in admin dashboard

alexBLR opened the pull request:
- Curate administrators can load preservation workflow metadata by submitting a CSV in the UI

- The work should happen in a background job

Resolves #1319

rotated8 commented on app/controllers/background_jobs_controller.rb:
```diff
@@ -0,0 +1,17 @@
+
+  def create
+    if params[:jobs] == 'cleanup'
+      FilesetCleanupService.new.execute
+    elsif params[:jobs] == 'preservation'
+      LoadPreservationMetadataService.new({ file: params[:preservation_csv] }).execute
+    else
+      ReindexFilesService.new({ file: params[:reindex_csv] }).execute
```
Rather than using a service, can we enqueue these jobs directly? What do you like about using the services?

alexBLR replied:
If we ever wanted to reuse it in a rake task to keep it DRY

rotated8 commented on config/routes.rb:
```diff
@@ -62,6 +62,8 @@
     end
   end
 
+  resources :background_jobs
```
As I understand it, `resources` means it creates paths for all the CRUD operations. Is that right? What happens if someone tries to get to background_job#index?

alexBLR replied:
You are right. I need to limit it to new and create.

rotated8 commented on app/views/hyrax/dashboard/sidebar/_tasks.html.erb:
```diff
@@ -23,6 +23,10 @@
     <span class="fa fa-upload"></span> <span class="sidebar-action-text">Import Content From a CSV</span>
   <% end %>
+
+  <%= menu.nav_link('/background_jobs/new',  data: { turbolinks: false }) do %>
+    <span class="fa fa-tasks"></span> <span class="sidebar-action-text">Background Jobs</span>
+  <% end %>
```
Nice!

rotated8 commented on spec/services/reindex_files_service_spec.rb:
```diff
@@ -0,0 +1,15 @@
+  describe 'execute' do
+    it 'loads preservation preservation metadata' do
+      described_class.new({ file: csv_file }).execute
+      expect(ReindexObjectJob).to have_been_enqueued.exactly(:twice)
```
Twice, because there are two ids in the CSV, right?

alexBLR replied:
Yes. 2 ids in the fixture

alexBLR commented:
Changes: removed services and service tests, put everything in the controller, wrote additional tests for the controller, limited routes for background jobs controller, added a dynamic flash message on create.
